apps/web/src/pages/Applications/ApplicationSelfDeclarationPage/ApplicationSelfDeclarationPage.tsx

```diff
@@ -0,0 +1,419 @@
+          <HelpLink />
+          <p data-h2-font-weight="base(700)" data-h2-margin="base(x1, 0)">
+            {intl.formatMessage(
+              {
+                defaultMessage:
+                  "See <whyLink>why we are asking you to self declare</whyLink>, <verificationLink>how this will be verified</verificationLink> and the term <definitionLink>Indigenous as defined for this program</definitionLink>.",
+                id: "AMboRG",
+                description:
+                  "Links to more information on the self-declaration process and definition of Indigenous",
+              },
+              {
+                whyLink,
+                verificationLink,
+                definitionLink,
+              },
+            )}
+          </p>
+        </form>
+      </FormProvider>
+    </>
+  );
+};
+
+const ApplicationSelfDeclarationPage = () => {
+  const intl = useIntl();
+  const paths = useRoutes();
+  const { applicationId } = useParams();
+  const [
+    {
+      data: applicationData,
+      fetching: applicationFetching,
+      error: applicationError,
+      stale: applicationStale,
+    },
+  ] = useGetApplicationQuery({
+    requestPolicy: "cache-first",
+    variables: {
+      id: applicationId || "",
+    },
+  });
+  const [{ data: userData, fetching: userFetching, error: userError }] =
+    useGetMeQuery();
+
+  const navigate = useNavigate();
+  const { followingPageUrl } = useApplicationContext();
+  const { applicantDashboard } = useFeatureFlags();
+  const [, executeMutation] = useUpdateUserAndApplicationMutation();
+  const cancelPath = applicantDashboard ? paths.dashboard() : paths.myProfile();
+  const nextStep = followingPageUrl ?? cancelPath;
+
+  const application = applicationData?.poolCandidate;
+  const resolvedIndigenousCommunities = unpackMaybes(
+    userData?.me?.indigenousCommunities,
+  );
+
+  const handleSubmit: SubmitHandler<FormValues> = async (formValues) => {
+    // Have to update both the user and the pool candidate in same request.  If you try to update just the user first and the application afterwards it interferes with the navigation.  I guess it creates a race condition as one of the contexts automatically refreshes.
+    executeMutation({
+      userId: userData?.me?.id || "",
+      userInput: {
+        indigenousDeclarationSignature: formValues.signature,
+        indigenousCommunities: formValuesToApiCommunities(formValues),
+      },
+      applicationId: applicationId || "",
+      applicationInput: {
+        insertSubmittedStep: ApplicationStep.SelfDeclaration,
+      },
+    })
+      .then((result) => {
+        if (result.error) throw new Error("Update user and application failed");
+
+        toast.success(
+          intl.formatMessage({
+            defaultMessage: "Successfully updated your profile",
+            id: "Ob5DQD",
+            description:
+              "Message displayed to users when saving profile is successful.",
+          }),
+        );
```

Comment:
Should this message be a little more contextual? 🤔 Something like "Successfully updated your Indigenous community(ies)"

Reply:
Well, it's not _wrong_. If we do that, we might want to do it everywhere.

Reply:
Yeah, I just found this one particularity confusing since there is also a step called "profile" 😛 To a user, it is not clear that indigenous community is also part of your profile until you get to that step.

Reply:
I guess we don't want to call it profile any more but what should it be?  "Successfully updated your self-declaration"?